[Li Po Hsien] iP

lpohsien opened the pull request:
# Duke
This is a simple to-do list manager.
## Features
Here are the commands you can use:
- add _task_
- remove _task_number_
- mark _task_number_
- unmark _task_number_
- list 

For the `add` commands, there are 3 types of tasks you can add to the to-do list. The possible tasks type are listed here:
1. **Todo**: A simple task with only a description.
2. **Deadline**: A simple task with a description and a due date.
3. **Event**: A simple task with a description, a start time, and an end time.

The encoding for the 3 task types is done based on the 
``` java
private String taskTypeString() {
    switch (this.type) {
    case TODO:
        return "T";
    case EVENT:
        return "E";
    case DEADLINE:
        return "D";
    default:
        return "";
    }
}
``` 
To use the manager, just follow these steps:
- [ ] Download the [jar file](https://github.com/lpohsien/ip/releases/tag/A-Jar)
- [ ] Launch it using `java -jar duke.jar`
- [ ] Start adding your tasks today! :+1:

> "Each day I will accomplish one thing on my to-do list." – Anonymous

Commit:
Add Gradle support

lpohsien committed:
Add user echoing

lpohsien committed:
Add ability to store text

lpohsien committed:
Fix multi-words input

lpohsien committed:
Add MarkableList to represent the items

lpohsien committed:
Split task into todo/deadline/event

lpohsien committed:
Modify input.text and EXPECTED.txt

lpohsien committed:
Add error handling

lpohsien committed:
Add delete option

lpohsien committed:
Add comments rename MarkableList to TaskList

lpohsien committed:
Add capability to save list

lpohsien committed:
Replace strings with LocalDateTime for datetime

lpohsien committed:
Merge branch 'branch-Level-7'
Add feature to save the list as a file and retrieve it on launch.

lpohsien committed:
Merge features from Level 7 and 8

lpohsien committed:
Refactor using OOP

lpohsien committed:
Package the tasks, commands and storage

lpohsien committed:
Add test cases

lpohsien committed:
Modify input.text and EXPECTED.txt

lpohsien committed:
Replace strings with LocalDateTime for datetime

lpohsien committed:
Refactor using OOP

lpohsien committed:
Resolve conflict:

lpohsien committed:
Package the tasks, commands and storage

lpohsien committed:
Add gradle support

lpohsien committed:
Fix bug

lpohsien committed:
Fix bugs in UI

lpohsien committed:
Add javadoc comments

lpohsien committed:
Fix code to follow coding standards

lpohsien committed:
Merge branch 'branch-A-Javadoc'

lpohsien committed:
Merge branch 'branch-A-CodingStandard'

lpohsien committed:
Add feature for find using keyword

lpohsien committed:
Merge branch 'branch-Level-9'

Devanshshah1309 commented on src/main/java/Duke/Command/Command.java:
```diff
@@ -0,0 +1,29 @@
+     * 
+     * @return True if this is the exit command. False otherwise.
+     */
+    public boolean isExit() {
+        return false;
+    } 
```
Hey, is there a reason the default implementation returns `false`? Might be better to just leave it as an abstract method right?

Devanshshah1309 commented on src/main/java/Duke/Command/DelCommand.java:
```diff
@@ -0,0 +1,40 @@
+
+/**
+ * This class represents the delete command that removes tasks
+ * from the todo list.
```
Can I check whether "todo list" refers to the "task list" instead?

Devanshshah1309 commented on src/main/java/Duke/Command/InvalidCommandException.java:
```diff
@@ -0,0 +1,12 @@
+ * This class represents the exceptions that occurs when an
+ * invalid command was given by the user.
+ */
+public class InvalidCommandException extends Exception{
```
Great naming for this haha, I really love the OOP design in general :)

Devanshshah1309 commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,133 @@
+     */
+    public static Command parse(String input) throws InvalidCommandException {
+        String errMsg = "";
+        if (input.startsWith("bye")) {
+            return new ExitCommand();
+        } else if (input.startsWith("list")) {
+            return new ListCommand();
+        } else if (input.startsWith("mark")) {
```
Would it be better to use a switch statement here? Not sure if that would be cleaner

Devanshshah1309 commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,133 @@
+     * @return The splitted string.
+     * @throws InvalidCommandException
+     */
+    private static String[] splitWithFormat(String input, String delimiter, String errMsg) 
```
Good abstraction here haha 👍 

Can possibly abstract away the max number of splits too (e.g. in this case, 2)

Devanshshah1309 reviewed (commented):
Awesome work in general! Great code design 😄

lpohsien committed:
Add GUI

lpohsien committed:
Add assertion.

Add assert statements to ensure that the assumptions are verified during
file reading and writing.

lpohsien committed:
Improve code quality.

Edit parser to adhere to cod quality guidelines more closely.

lpohsien committed:
Merge pull request #1 from lpohsien/branch-A-CodeQuality

Branch A-CodeQuality

lpohsien committed:
Merge from master

lpohsien committed:
Merge pull request #2 from lpohsien/branch-Level-10

Add GUI feature from branch-Level-10

lpohsien committed:
Fix GUI exit

lpohsien committed:
Add mass ops option for delete, mark and unmark

lpohsien committed:
Merge pull request #3 from lpohsien/branch-C-MassOps

Add mass ops option for delete, mark and unmark

lpohsien committed:
Improve UI

lpohsien committed:
Add credit for GUI adaptation

lpohsien committed:
Fix clipping of messages

lpohsien committed:
Add window title

lpohsien committed:
Add product screenshot

lpohsien committed:
Add product description

lpohsien committed:
Update feature list

lpohsien committed:
Remove indexing for operation output

lpohsien committed:
Add usage

lpohsien committed:
Implement task indexing for find command.

lpohsien committed:
Update README with full usage guide

lpohsien committed:
Update README

lpohsien committed:
Resolve issue #4

lpohsien committed:
Fix Readme
